src/clients/memberclient.cpp

```diff
@@ -106,6 +111,41 @@ void display(const json& proposals)
   }
 }
 
+uint8_t hex_char_to_byte(char c)
+{
+  if (c >= '0' && c <= '9')
+  {
+    return c - '0';
+  }
+  if (c >= 'A' && c <= 'F')
+  {
+    return c - 'A' + 0xa;
+  }
+  if (c >= 'a' && c <= 'f')
+  {
+    return c - 'a' + 0xa;
+  }
+  return -1;
+}
+
+template <size_t SZ>
+void hex_str_to_bytes(const std::string& src, std::array<uint8_t, SZ>& dst)
+{
+  if (src.length() != SZ << 1)
+  {
+    throw logic_error("Invalid code id length");
+  }
+  auto hex_str = src.c_str();
+
+  for (size_t i = 0; i < SZ; ++i)
```

Comment:
Rather than writing our own `hex_char_to_bytes`, could use [`strtoul`](http://www.cplusplus.com/reference/cstdlib/strtoul/) with base 16.

It's a bit awkward because it requires null-terminated byte-pairs and a cast, something like:
```
dst[i] = static_cast<uint8_t>(strtoul(src.substr(i*2, 2).c_str(), nullptr, 16));
```

Reply:
I suppose we can create 2 byte null-terminated strings, I really don't have any preference.